src/test/xmeta/test_symbol_table.cpp

```diff
@@ -517,3 +515,377 @@ TEST_CASE("Resolving types across namespaces test")
+            {
+            };
+
+            interface IControl
+            {
+                S1 property1;
+                M.S2 property2;
+            }
+        }
+        namespace M
+        {
+            struct S2
+            {
+            };
+        }
+    )" };
+
+    xmeta_idl_reader reader{ "" };
+    reader.read(test_idl);
+    REQUIRE(reader.get_num_syntax_errors() == 0);
+
+    auto const& namespaces = reader.get_namespaces();
+    auto const& it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto const& ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto const& interfaces = ns_bodies[0]->get_interfaces();
+    REQUIRE(interfaces.size() == 1);
+
+    REQUIRE(interfaces.find("IControl") != interfaces.end());
+    auto const& model = interfaces.at("IControl");
+
+    auto const& properties = model->get_properties();
+    {
+        REQUIRE(properties[0]->get_id() == "property1");
+        auto const& property_type = properties[0]->get_type().get_semantic();
+        REQUIRE(property_type.is_resolved());
+        REQUIRE(std::get<std::shared_ptr<struct_model>>(property_type.get_resolved_target())->get_id() == "S1");
+    }
+    {
+        REQUIRE(properties[1]->get_id() == "property2");
+        auto const& property_type = properties[1]->get_type().get_semantic();
+        REQUIRE(property_type.is_resolved());
+        REQUIRE(std::get<std::shared_ptr<struct_model>>(property_type.get_resolved_target())->get_id() == "S2");
+    }
+}
+
+TEST_CASE("Interface event test")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            delegate void StringListEvent(Int32 sender);
+            interface IControl
+            {
+                event StringListEvent Changed;
+            }
+        }
+    )" };
+    std::vector<std::string> paths = { "Foundation.xmeta" };
+    xmeta_idl_reader reader{ "" , paths};
+    reader.read(test_idl);
+    REQUIRE(reader.get_num_syntax_errors() == 0);
+
+    auto const& namespaces = reader.get_namespaces();
+    auto const& it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto const& ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto const& interfaces = ns_bodies[0]->get_interfaces();
+    REQUIRE(interfaces.size() == 1);
+
+    REQUIRE(interfaces.find("IControl") != interfaces.end());
+    auto const& model = interfaces.at("IControl");
+
+    auto const& events = model->get_events();
+    REQUIRE(events[0]->get_id() == "Changed");
+
+    auto const& property_type = events[0]->get_type().get_semantic();
+    REQUIRE(property_type.is_resolved());
+    REQUIRE(std::holds_alternative<std::shared_ptr<delegate_model>>(property_type.get_resolved_target()));
+    REQUIRE(std::get<std::shared_ptr<delegate_model>>(property_type.get_resolved_target())->get_id() == "StringListEvent");
+}
+
+TEST_CASE("Interface circular inheritance test")
```

Comment:
I alluded to this above, but I'd like to see more tests showing that we properly reject invalid IDL.